core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -728,6 +736,23 @@ private[spark] class BlockManager(
     }
   }
 
+  /**
+   * Get the local merged shuffle block data for the given block ID as multiple chunks.
+   * A merged shuffle file is divided into multiple chunks according to the index file.
+   * Instead of reading the entire file as a single block, we split it into smaller chunks
+   * which will be memory efficient when performing certain operations.
+   */
+  override def getLocalMergedBlockData(blockId: ShuffleBlockId): Seq[ManagedBuffer] = {
```

Comment:
This is also the older API. @zhouyejoe We need to add the dirs. Please look at these APIs in https://github.com/apache/spark/pull/32140/